clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -1121,13 +1140,25 @@ public long maybeUpdate(long now) {
 
             // Beware that the behavior of this method and the computation of timeouts for poll() are
             // highly dependent on the behavior of leastLoadedNode.
-            Node node = leastLoadedNode(now);
-            if (node == null) {
+            LeastLoadedNode leastLoadedNode = leastLoadedNode(now);
+
+            // Rebootstrap if needed and configured.
+            if (!leastLoadedNode.hasNodeAvailableOrConnectionReady()
+                    && metadataRecoveryStrategy == MetadataRecoveryStrategy.REBOOTSTRAP) {
```

Comment:
Should we swap this around to `if (metadataRecoveryStrategy == MetadataRecoveryStrategy.REBOOTSTRAP && !leastLoadedNode.hasNodeAvailableOrConnectionReady())` here and in the admin client, so that we only use the new method if rebootstrap is configured?

Reply:
As it's only comparisons inside the method, performance-wise it doesn't matter much. However, it's more logical. Done this.